UnboundLocalError when using upload method

Hi,
I've got this error 

`FAILED config/tests/test_class_storage.py::TestStorage::test_upload_file[bristen-/capstor/scratch/cscs/firecr01/f7t_periodic_tests/tests_prod_20250825T091658-a-csstaff] - UnboundLocalError: cannot access local variable 'resp' where it is not associated with a value`

when using pyfirecrest in this way:

```
filename = os.path.basename(source_file)
ext_upload = f7t_client.upload(system,
                               local_file=source_file,
                               directory=directory,
                               filename=filename,
                               account=account,
                               blocking=False)

if ext_upload:
    ext_upload.upload_file_to_stage()
    ext_upload.wait_for_transfer_job()
```

The last working version was 3.2.2. Both newer versions 3.3.0 and 3.3.1 are affected.
Therefore, I suppose the issue was introduced in 3.3.0 and it should be related to the **upload** method.

Thank you.
Kind Regards.

--- a/firecrest/v2/_sync/Client.py
+++ b/firecrest/v2/_sync/Client.py
@@ -117,16 +117,16 @@ class ExternalUpload:
         else:
             content_length = chunk_size
 
-            with open(self._local_file, "rb") as f:
-                f.seek(start)
-                resp = self._client._session.put(
-                    url=url,
-                    content=chunk_reader(f, self.chunk_size),
-                    timeout=None,
-                    headers={
-                        "Content-Length": str(content_length)
-                    }
-                )
+        with open(self._local_file, "rb") as f:
+            f.seek(start)
+            resp = self._client._session.put(
+                url=url,
+                content=chunk_reader(f, self.chunk_size),
+                timeout=None,
+                headers={
+                    "Content-Length": str(content_length)
+                }
+            )
 
         if resp.status_code >= 400:
             raise MultipartUploadException(

